engine/src/main/java/com/datatorrent/stram/api/StramEvent.java

```diff
@@ -493,6 +514,20 @@ public void setErrorMessage(String errorMessage)
       this.errorMessage = errorMessage;
     }
 
+    public String getLogFileName()
```

Comment:
I disagree. Every time an event is generated, there may be log entry associated with the event in the future. How those 14-15 classes will be affected. Should not all functionality be encapsulated in the parent class?